### PR TITLE
AR-4337 Update app list response

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-4337](https://team-1624093970686.atlassian.net/browse/AR-4337).

## Changes

- Update the response of app list (/user-app api)
- Maintain stats/overview in store
- Update dashboard page to show dynamic storage and bandwidth values
- (Addon) Wrap Create app inputs in a flex container

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/pages/AppDashboard.vue [hunk 1]
```diff
@@ -121,34 +121,39 @@ async function fetchAndPopulateCharts() {
   updateChart(periodicUsage.data)
 }
 
+function getAllowedLimit(limit: number) {
+  return limit || MAX_ALLOWED_APP_LIMIT_IN_BYTES
```
Comment (ajithranka) [inline src/pages/AppDashboard.vue:125]: When will `data.storage` and `data.bandwidth` be missing from the api response?

File: src/pages/ManageApps.vue [hunk 2]
```diff
@@ -135,15 +130,19 @@ appsStore.$subscribe(() => {
                 <VSeperator vertical class="stats-separator" />
                 <VStack direction="column" align="center">
                   <span class="stats-title">Estimated Cost</span>
-                  <span class="stats-number">$0</span>
+                  <span class="stats-number"
+                    >${{ app.estimatedCost || 0 }}</span
```
Comment (ajithranka) [inline src/pages/ManageApps.vue:134]: Is `estimatedCost` missing in any case from the api response?

File: src/stores/apps.store.ts [hunk 3]
```diff
@@ -77,25 +77,49 @@ type App = {
...
+  totalUsers: number
+  createdAt: string
+}
+
 type AppState = {
   appIds: AppId[]
   appsById: {
     [key: AppId]: App
   }
+  appsOverview: {
```
Comment (ajithranka) [inline src/stores/apps.store.ts:102]: Maybe we should all this `appsOverviewById` to keep the same convention as `appsById`?

Comment (shrinathprabhu) [reply src/stores/apps.store.ts]: Resolved at https://github.com/arcana-network/developer-dashboard/pull/135/commits/9e785654fea8c963d6010cff341b8ceaa7fe2278

File: src/pages/ManageApps.vue (hunk 2 above)
Comment (shrinathprabhu) [reply src/pages/ManageApps.vue]: Removed here https://github.com/arcana-network/developer-dashboard/pull/135/commits/134e776175d244327e99ab2630bdd290597ff539

File: src/pages/AppDashboard.vue (hunk 1 above)
Comment (shrinathprabhu) [reply src/pages/AppDashboard.vue]: Removed here https://github.com/arcana-network/developer-dashboard/pull/135/commits/134e776175d244327e99ab2630bdd290597ff539

Comment (shrinathprabhu) [reply src/pages/AppDashboard.vue]: @ajithranka no, I just confirmed, these fields won't be missing from api response, it'll always be there

File: src/pages/ManageApps.vue (hunk 2 above)
Comment (shrinathprabhu) [reply src/pages/ManageApps.vue]: @ajithranka no, I just confirmed, estimated_cost won't be missing from api response, it'll always be there

File: src/pages/AppDashboard.vue (hunk 1 above)
Comment (ajithranka) [reply src/pages/AppDashboard.vue]: Then I think we can remove the fallback limit from the frontend.

File: src/pages/ManageApps.vue (hunk 2 above)
Comment (ajithranka) [reply src/pages/ManageApps.vue]: We can remove `|| 0` in that case.

**Review by ajithranka — APPROVED** [review]

File: src/components/app-configure/CreateApp.vue [hunk 4]
```diff
@@ -49,6 +49,7 @@ async function handleCreateApp() {
       region: RegionMapping[selectedRegion.value.value],
     })
   ).data.app
+  const appOverview = (await fetchStats(app.ID)).data
```
Comment (karthik-durai) [inline src/components/app-configure/CreateApp.vue:52]: looks like if `fetchStats` throws error, the app might crash

File: src/components/app-configure/auth/WebWallet.vue [hunk 5]
```diff
@@ -78,14 +78,17 @@ function hasSameValuesInStore() {
 
 async function handleSave() {
   try {
-    const { auth } = app
-    auth.wallet.websiteDomain = walletWebsiteDomain.value
-    auth.wallet.walletType = hasUIMode.value ? WalletMode.UI : WalletMode.NoUI
-    auth.wallet.selectedTheme = selectedTheme.value.value
+    const { auth } = { ...app }
```
Comment (karthik-durai) [inline src/components/app-configure/auth/WebWallet.vue:81]: why are spreading before de-structuring?

Comment (shrinathprabhu) [reply src/components/app-configure/auth/WebWallet.vue]: Yeah actually not required. Removed [here](https://github.com/arcana-network/developer-dashboard/pull/135/commits/7ace58432fb05f1afdc58d173ea1be8c9918b9ab)

File: src/components/app-configure/CreateApp.vue (hunk 4 above)
Comment (shrinathprabhu) [reply src/components/app-configure/CreateApp.vue]: Good point. Wrapped the entire flow in try catch and showing toast if error occurs. Resolved [here](https://github.com/arcana-network/developer-dashboard/pull/135/commits/7ace58432fb05f1afdc58d173ea1be8c9918b9ab)

**Review by karthik-durai — APPROVED** [review]

**Pull request merged**